project_from_quotation/models/sale_order.py

```diff
@@ -0,0 +1,193 @@
+        """
+        super()._compute_tasks_ids()
+        for order in self:
+            quotation_tasks = self.env["project.task"].search(
+                [
+                    ("quotation_line_id", "in", order.order_line.ids),
+                ]
+            )
+            order.tasks_ids |= quotation_tasks
+            order.tasks_count = len(order.tasks_ids)
+        return
+
+    @api.model
+    def _prepare_task_vals(
+        self,
+        project,
+        order_line,
+    ):
+        """
+        Prepare vals for task creation
+
+        Args:
+        project (project.project): Project
+        order_line (sale.order.line): Order line
+
+        Returns:
+            dict: Dictionary of task values
+        """
+        return {
+            "name": order_line.name,
+            "project_id": project.id,
+            "quotation_line_id": order_line.id,
+            "active": True,
+        }
+
+    @api.model
+    def _prepare_project_vals(self, sale_order):
+        """
+        Prepare vals for project creation
+
+        Args:
+        sale_order (sale.order): Sale Order
+
+        Returns:
+            dict: Dictionary of project values
+        """
+        return {
+            "name": sale_order.name,
+            "quotation_id": sale_order.id,
+            "active": True,
+        }
+
+    def _get_order_lines_unique_products(self, order_lines=None):
+        """
+        Get order lines with unique products
+
+        Args:
+            order_lines (sale.order.line recordset): Order lines
+            if not set, uses self.order_line
+
+        Returns:
+            sale.order.line recordset: Every first
+            line with the same product
+        """
+        self.ensure_one()
+        unique_products = []
+        res_order_lines = self.env["sale.order.line"]
+        order_lines = (order_lines or self.order_line).filtered("product_id")
+
+        for line in order_lines:
+            if line.product_id.id not in unique_products:
+                unique_products.append(line.product_id.id)
+                res_order_lines |= line
+
+        return res_order_lines
+
+    def _create_project_from_quotation(
+        self,
+        mode="project",
+        order_lines=None,
+    ):
+        """
+        Create project from a quotation
+
+        Args:
+        mode (str): Project creation mode. Possible values:
+            project - create a new empty project
+            tasks_per_line - create a project with tasks for each order line
+            tasks_per_product - create a project with tasks for each product
+        order_lines (sale.order.line): Order lines to use for tasks creation
+
+        Returns:
+            Created record of the project.project model
+        """
+        self.ensure_one()
+        project = False
+        task_obj = self.env["project.task"]
+        project_obj = self.env["project.project"]
+        order_lines = (order_lines or self.order_line).filtered(lambda x: not x.task_id)
```

Comment:
Same here. If some lines were removed on purpose we shouldn't use them in out flow